lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -677,33 +698,18 @@ unsafe fn from_go_array(len: u32, ptr: *mut u8) -> Vec<u8> {
     slice::from_raw_parts(ptr, len as usize).to_vec()
 }
 
-/// CGOError is an alias for a C string pointer, for C API clarity.
-pub type CGOError = *mut c_char;
-
-/// CGO_OK is a CGOError value that means "success".
-const CGO_OK: CGOError = ptr::null_mut();
-
-fn to_cgo_error(s: String) -> CGOError {
-    CString::new(s).expect("CString::new failed").into_raw()
-}
-
-/// from_cgo_error copies CGOError into a String and frees the underlying Go memory.
-///
-/// # Safety
-///
-/// The pointer inside the CGOError must point to a valid null terminated Go string.
-unsafe fn from_cgo_error(e: CGOError) -> String {
-    let s = from_go_string(e);
-    free_go_string(e);
-    s
+#[repr(C)]
+#[derive(Copy, Clone, PartialEq)]
+pub enum CGOErrCode {
+    ErrCodeSuccess,
```

Comment:
Can you add numeric values here to ensure the `ErrCodeSuccess` gets the value 0, which is traditional for C-based APIs to indicate no error.

Reply:
I believe `#[repr(C)]` gives us that here per C's default behavior, no harm making it explicit though.